err.log on windows:
~~~
gawk: cmd. line:1:  {last_section=section; section=$ 3}; /^COFF SYMBOL TABLE/{for(i in hide) delete hide[i]}; /Section length .*#relocs.*(pick any)/{hide[last_section]=1}; /^ *Symbol name *: /{split($ 0,sn,:); si=substr(sn[2],2)}; /^ *Type *: code/{print T,si,substr(si,length(prfx))}; /^ *Type *: data/{print I,si,substr(si,length(prfx))}; $ 0!~/External *\|/{next}; / 0+ UNDEF /{next}; / UNDEF \([^|]\)*()/{next}; {if(hide[section]) next}; {f=D}; $ 0~/\(\).*\|/{f=T}; {split($ 0,a,/\||\r/); split(a[2],s)}; s[1]~/^[@?]/{print f,s[1],s[1]; next}; s[1]~prfx {split(s[1],t,@); print f,t[1],substr(t[1],length(prfx))} 
gawk: cmd. line:1:                                                                                                                                                                                            ^ syntax error
~~~
and out.log:
~~~
/bin/sh ../libtool  --tag=CC   --mode=compile compile cl.exe -DHAVE_CONFIG_H -DNO_XMALLOC -I. -I..  -I. -I. -I.. -I.. -DIN_LIBUNISTRING -DDEPENDS_ON_LIBICONV=1 -DWIN32 -D_WINDOWS -D_DEBUG  -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1 -c -o unistd.lo unistd.c
D:\buildtrees\libunistring\x64-windows-dbg\lib\config.h(8): error C2007: #define syntaxD:\buildtrees\libunistring\x64-windows-dbg\lib\config.h(8): error C2007: #define syntax
D:\buildtrees\libunistring\x64-windows-dbg\lib\config.h(9): error C2008: '.': unexpected in macro definitionD:\buildtrees\libunistring\x64-windows-dbg\lib\config.h(9): error C2008: '.': unexpected in macro definition
D:\buildtrees\libunistring\x64-windows-dbg\lib\config.h(10): error C2008: '.': unexpected in macro definitionD:\buildtrees\libunistring\x64-windows-dbg\lib\config.h(10): error C2008: '.': unexpected in macro definition
...
~~~

I didn't really plan to support windows -  at least it is not relevant neither for libidn2 nor for IDN support in curl. Unless it is resolved easily, I will mark windows as unsupported and move on to libidn2.

--- a/ports/libunistring/portfile.cmake
+++ b/ports/libunistring/portfile.cmake
@@ -1,6 +1,10 @@
 set(LIBUNISTRING_VERSION 0.9.10)
 set(LIBUNISTRING_FILENAME libunistring-${LIBUNISTRING_VERSION}.tar.gz)
 
+if(NOT VCPKG_TARGET_IS_MINGW)
+    vcpkg_fail_port_install(ON_TARGET "Windows" "UWP")
+endif()
+
 vcpkg_download_distfile(ARCHIVE
     URLS
         "https://ftp.gnu.org/gnu/libunistring/${LIBUNISTRING_FILENAME}"

--- a/ports/libunistring/vcpkg.json
+++ b/ports/libunistring/vcpkg.json
@@ -3,6 +3,7 @@
   "version": "0.9.10",
   "description": "GNU libunistring provides functions for manipulating Unicode strings and for manipulating C strings according to the Unicode standard.",
   "homepage": "https://www.gnu.org/software/libunistring/",
+  "supports": "!windows",
   "dependencies": [
     "libiconv"
   ]

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3668,6 +3668,10 @@
       "baseline": "4.3-0",
       "port-version": 0
     },
+    "libunistring": {
+      "baseline": "0.9.10",
+      "port-version": 0
+    },
     "liburing": {
       "baseline": "2.0",
       "port-version": 0

--- a/versions/l-/libunistring.json
+++ b/versions/l-/libunistring.json
@@ -1,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "586bf47f6b9ee55aa3eb9ec7c7ff36363e39980b",
+      "version": "0.9.10",
+      "port-version": 0
+    }
+  ]
+}

